Add Swing "Chat" Feature to WebUI

TheBestPessimist opened the pull request:
https://github.com/metasfresh/metasfresh/issues/6504

teosarca commented on de.metas.adempiere.adempiere/base/src/main/java/de/metas/notes/NotesRepository.java:
```diff
@@ -0,0 +1,115 @@
+	}
+
+	@NonNull
+	public List<I_CM_ChatEntry> retrieveNotes(@NonNull final TableRecordReference tableRecordReference, final int maxNumberOfRecords)
```
pls introduce a POJO to be returned instead of I_CM_ChatEntry

teosarca replied:
rename the method to `retrieveLastNotes`

TheBestPessimist replied:
done

teosarca commented on de.metas.adempiere.adempiere/base/src/main/java/de/metas/notes/NotesRepository.java:
```diff
@@ -0,0 +1,115 @@
+	}
+
+	@NonNull
+	private I_CM_Chat getOrCreateChat(final @NonNull TableRecordReference tableRecordReference)
```
pls introduce a POJO to be returned instead of I_CM_Chat

TheBestPessimist replied:
Fixed by returning `ChatId` since that's what i really need.

TheBestPessimist commented on de.metas.adempiere.adempiere/base/src/main/java-legacy/org/adempiere/ad/dao/impl/TypedSqlQuery.java:
```diff
@@ -515,12 +515,6 @@ else if (postQueryFilter.accept(model))
-	 * @return first ID or -1 if not found
-	 * @throws DBException
-	 */
 	@Override
```
This is the implementation class. I removed the comments from here and added better comments in the interface

TheBestPessimist commented on de.metas.adempiere.adempiere/base/src/main/java/de/metas/comments/ChatEntryId.java:
```diff
@@ -0,0 +1,59 @@
+	int repoId;
+
+	@JsonCreator
+	public static ChatEntryId ofRepoId(final int repoId)
```
Sounds like a good idea. Will implement.

TheBestPessimist replied:
No longer implementing this. Explanation below.

TheBestPessimist commented on de.metas.adempiere.adempiere/base/src/main/java/de/metas/comments/CommentsRepository.java:
```diff
@@ -0,0 +1,132 @@
+
+	public void createComment(final @NonNull String characterData, @NonNull final TableRecordReference tableRecordReference)
+	{
+		final ChatId chatId = getOrCreateChat(tableRecordReference);
```
Instead of ChatId/ChatEntryId? What about the tables/POs?

TheBestPessimist replied:
The POs and tables will not be renamed.

TheBestPessimist commented:
Discussed with teo. `ChatId` is pretty useles. I have renamed it to `CommentEntryParentId` and added a comment explaining that it's pretty useless and that `CommentEntryId` should be used instead.
